ResearchKit/Consent/ORKVisualConsentStepViewController.m

```diff
@@ -251,10 +252,30 @@ - (IBAction)goToPreviousPage {
 }
 
 - (IBAction)next {
-    [self showViewController:[self viewControllerForIndex:[self currentIndex]+1] forward:YES animated:YES];
-    ORKAccessibilityPostNotificationAfterDelay(UIAccessibilityScreenChangedNotification, nil, 0.5);
+    ORKConsentSceneViewController *currentConsentSceneViewController = [self viewControllerForIndex:[self currentIndex]];
+    [currentConsentSceneViewController setScrollEnabled:NO];
+    if ([currentConsentSceneViewController isScrolledToTop])
+    {
+        [self showNextViewController];
+    }
+    else
+    {
+        [currentConsentSceneViewController setScrollViewDelegate:self];
+        [currentConsentSceneViewController scrollToTopAnimated:YES];
+    }
 }
 
+- (void)scrollViewDidEndScrollingAnimation:(UIScrollView *)scrollView {
+    self.delegate = nil;
```

Comment:
It's just defensive coding in case the user goes backward and forward again. Not clearing it should not be a problem, but the `delegate` has no use after the scrolling animation ends, and it's re-set before the `scrollToTopAnimated:` method is called anyway.